### PR TITLE
Only allow at most 20 new models to be pending eval.

### PR DESCRIPTION
**Pull request by Sid-Data-Universe**

### CODE REVIEW

File: neurons/validator.py [hunk 1]
```diff
@@ -348,6 +348,26 @@ def update_models(self):
...
+                    pending_uid_count = len(self.pending_uids_to_eval)
+                    current_uid_count = len(self.uids_to_eval)
+
+                # Only allow at most 20 pending uids + sample min (for new vali startup).
+                while (
+                    pending_uid_count + current_uid_count >= 20 + self.config.sample_min
+                ):
+                    # Wait 5 minutes for the eval loop to process them.
+                    bt.logging.info(
+                        f"Update loop: Already 20 synced models pending eval. Waiting 5 minutes."
```
Comment (RusticLuftig) [inline neurons/validator.py:364]: nit. Can we say, "checking again in 5 minutes". I fear people will see "waiting" and wonder why we pause for 5 minutes

File: neurons/validator.py [hunk 2]
```diff
@@ -348,6 +348,26 @@ def update_models(self):
...
+                    )
+                    time.sleep(300)
+                    # Check to see if the pending uids have been cleared yet.
+                    with self.pending_uids_to_eval_lock:
+                        pending_uid_count = len(self.pending_uids_to_eval)
```
Comment (RusticLuftig) [inline neurons/validator.py:369]: Do we not need to also check the current_uid_count again here?

**Review by RusticLuftig — CHANGES_REQUESTED** [review]

File: neurons/validator.py (hunk 1 above)
Comment (Sid-Data-Universe) [reply neurons/validator.py:364]: Done

File: neurons/validator.py (hunk 2 above)
Comment (Sid-Data-Universe) [reply neurons/validator.py:369]: good catch.

**Review by RusticLuftig — APPROVED** [review]

**Pull request merged**